### ISSUE TITLE
Firefox not finding org.freedesktop.portal.Settings

### ISSUE THREAD

**Issue by Arxcis** [emersion/xdg-desktop-portal-wlr#167]
Hi. 

- I am on a completely fresh install of Arch Linux + Sway.
- I followed this guide to install Arch https://www.youtube.com/watch?v=PQgyW10xD8s from DistroTube
- After completing the guide, I installed the following:
```
pacman -S sway alacritty
pacman -S flatpak xdg-desktop-portal-wlr wireplumber
pacman -S wofi
```
- After rebooting I launched into sway by running:
```
sway
flatpak install firefox
wofi --show drun (then select firefox from the launcher)
```
- Firefox opens but does give warnings in the command-line that it cannot find portals and it. Example:
```
No such interface ?org.freedesktop.portal.Settings? on object at path /org/freedesktop/portal/desktop
```
This is apparent, because dark-theme is not correctly loaded from settings, and filechooser-dialog does not work, when xdg-desktop-portal flag is enabled in firefox about:config.

- When I do `systemctl --user` I have the following services listed:
```
dbus.service                      loaded active running
flatpak-session-helper.service   loaded active running
pipewire.service                     loaded active running
wireplumber.service                     loaded active running
xdg-desktop-portal-wlr.service     loaded active running
xdg-desktop-portal.service     loaded active running
xdg-document-portal.service     loaded active running
xdg-permission-store.service    loaded active running
```

Everything looks good to me. Where am I wrong?

**Comment by emersion** [comment]
Closing because this isn't an xdpw bug, it doesn't provide `org.freedesktop.portal.Settings`. xdg-desktop-portal-gtk is supposed to I think?

**Comment by Arxcis** [comment]
> Closing because this isn't an xdpw bug, it doesn't provide `org.freedesktop.portal.Settings`. xdg-desktop-portal-gtk is supposed to I think?

Alright. Thank you for a very quick answer :pray: 

So when starting on a fresh system, Am I supposed to installed multiple `xdg-desktop-portal`-backends to get support for all the portals?
Do they play nice together?
Could I do:
```
sudo pacman -S xdg-desktop-portal-wlr xdg-desktop-portal-gtk xdg-desktop-portal-gnome xdg-desktop-portal-kde
```
And it would work ok?

**Comment by emersion** [comment]
Yes, installing multiple backends at the same time should work fine. For wlroots-based compositors, you'll probably need -wlr and -gtk only.

**Comment by Arxcis** [comment]
> Yes, installing multiple backends at the same time should work fine. For wlroots-based compositors, you'll probably need -wlr and -gtk only.

I did not know about this. Thank you for enlightening me. Do you by any chance have a link to a resource where I can read more about this? I found it difficult to search for this topic.
 
> Backends a prioritized depending on `XDG_CURRENT_DESKTOP`. As last resort, an arbitrary backend is picked.

OK. I installed `xdg-desktop-portal-gtk` (in addition to the already installed `xdg-desktop-portal-wlr`) and set `XDG_CURRENT_DESKTOP=sway` in `$HOME/.bashrc`.

Now the warnings are gone in Firefox, and Flatseal was able to pick up the correct dark theme :+1:
